tony-core/src/main/java/com/linkedin/tony/util/ParserUtils.java

```diff
@@ -59,12 +61,19 @@
    */
   @VisibleForTesting
   public static boolean isValidHistFileName(String fileName, String jobIdRegex) {
-    String histFileNoExt = fileName.substring(0, fileName.lastIndexOf('.'));
+    String histFileNoExt = fileName.substring(0, fileName.indexOf('.'));
     String[] metadataArr = histFileNoExt.split("-");
-    if (metadataArr.length != 5) {
+    if (metadataArr.length < 3) {
       LOG.error("Missing fields in metadata");
       return false;
     }
+
+    if (fileName.endsWith(Constants.INPROGRESS)) {
+      return metadataArr[0].matches(jobIdRegex)
+          && metadataArr[1].matches("\\d+")
+          && metadataArr[2].matches(metadataArr[2].toLowerCase());
+    }
+
     return metadataArr[0].matches(jobIdRegex)
         && metadataArr[1].matches("\\d+")
         && metadataArr[2].matches("\\d+")
```

Comment:
Please do a `metadataArr.length == 5` check to avoid `ArrayOutOfBoundsException`.

Reply:
👍